tests/unit/tasks/server/middleware/tests-server-test.js

```diff
@@ -47,5 +57,37 @@ describe('TestServerAddon', function () {
         }
       });
     });
+
+    it('allows baseURL containing `+` character', function(done) {
+      mockRequest.path = '/braden/+/tests/any-old-file';
+      mockRequest.headers.accept = ['*/*'];
+      addon.serverMiddleware({
+        app: app,
+        options: {
+          watcher: Promise.resolve({ directory: 'nothing' }),
+          baseURL: '/braden/+'
+        },
+        finally: function() {
+          expect(mockRequest.url).to.equal('/braden/+/tests/index.html');
+          done();
+        }
+      });
+
+      it('allows rootURL containing `+` character', function(done) {
+        mockRequest.path = '/grayson/+/tests/any-old-file';
+        mockRequest.headers.accept = ['text/html'];
+        addon.serverMiddleware({
+          app: app,
+          options: {
+            watcher: Promise.resolve({directory: 'nothing'}),
+            rootURL: '/grayson/+'
+          },
+          finally: function () {
+            expect(mockRequest.url).to.equal('/grayson/+/tests/index.html');
+            done();
+          }
+        });
+      });
```

Comment:
I could also test some negative cases. I chose not to go crazy with testing (I've been known to do that). Let me know if you think we should add a few more cases (e.g., it should not modify the url if the file exists, if it doesn't match the test pattern, etc.).